runner/src/server/plugins/engine/components/ComponentCollection.ts

```diff
@@ -71,7 +71,17 @@ export class ComponentCollection {
   getPrePopulatedItems() {
     return this.formItems
       .filter((item) => item.options?.allowPrePopulation)
-      .map((item) => item.getStateSchemaKeys())
+      .map((item) => {
+        const schema = item.getStateSchemaKeys();
+        const schemaMapping = Object.entries(schema)[0];
+        return {
+          [schemaMapping[0]]: {
+            schema: schemaMapping[1],
+            allowOverwriteFromQueryParam:
+              item.options.allowOverwriteFromQueryParam,
+          },
+        };
+      })
```

Comment:
The array indexing makes it hard to follow

```suggestion
        return {
          [item.name]: {
            schema: schema[item.name],
            allowOverwriteFromQueryParam:
              item.options.allowOverwriteFromQueryParam,
          },
        };
      })
```

alternatively, you can use destructuring to give schemaMapping[0,1] more meaningful names, but  `Object.entries(schema)[0];` still looks a bit odd imo

```suggestion
        const [key, joiSchema] = Object.entries(schema)[0];
        return {
          [key]: {
            schema: joiSchema,
            allowOverwriteFromQueryParam:
              item.options.allowOverwriteFromQueryParam,
          },
        };
      })
```